consensus/istanbul/core/core.go

```diff
@@ -362,63 +378,19 @@ func (c *core) commit() error {
 	c.backlog.updateState(c.current.View(), c.current.State())
 
 	proposal := c.current.Proposal()
-	if proposal != nil {
-		aggregatedSeal, err := GetAggregatedSeal(c.current.Commits(), c.current.Round())
-		if err != nil {
-			nextRound := new(big.Int).Add(c.current.Round(), common.Big1)
-			logger.Warn("Error on commit, waiting for desired round", "reason", "getAggregatedSeal", "err", err, "desired_round", nextRound)
-			c.waitForDesiredRound(nextRound)
-			return nil
-		}
-		aggregatedEpochValidatorSetSeal, err := GetAggregatedEpochValidatorSetSeal(proposal.Number().Uint64(), c.config.Epoch, c.current.Commits())
-		if err != nil {
-			nextRound := new(big.Int).Add(c.current.Round(), common.Big1)
-			c.logger.Warn("Error on commit, waiting for desired round", "reason", "GetAggregatedEpochValidatorSetSeal", "err", err, "desired_round", nextRound)
-			c.waitForDesiredRound(nextRound)
-			return nil
-		}
-		if err := c.backend.Commit(proposal, aggregatedSeal, aggregatedEpochValidatorSetSeal); err != nil {
-			nextRound := new(big.Int).Add(c.current.Round(), common.Big1)
-			logger.Warn("Error on commit, waiting for desired round", "reason", "backend.Commit", "err", err, "desired_round", nextRound)
-			c.waitForDesiredRound(nextRound)
-			return nil
-		}
-	}
-
-	logger.Info("Committed")
-	return nil
-}
-
-// GetAggregatedEpochValidatorSetSeal aggregates all the given seals for the SNARK-friendly epoch encoding
-// to a bls aggregated signature. Returns an empty signature on a non-epoch block.
-func GetAggregatedEpochValidatorSetSeal(blockNumber, epoch uint64, seals MessageSet) (types.IstanbulEpochValidatorSetSeal, error) {
-	if !istanbul.IsLastBlockOfEpoch(blockNumber, epoch) {
-		return types.IstanbulEpochValidatorSetSeal{}, nil
+	if proposal == nil {
+		return nil
```

Comment:
It seems odd that a `nil` proposal can be committed. Unless there is an explanation for why this would happen during normal operation, I'd like us to add a warning here.
```suggestion
		logger.Warn("Committed nil proposal")
		return nil
```

Reply:
I think proposal can be nil if the node never receives the pre-prepare, though it also should get stuck on the accept pre-prepare stage

Reply:
IIUC it would still be a bug if that were to happen, correct?

Reply:
If we tried to commit a nil proposal, I think it's a bug.

Reply:
I would be in favour of panicking here again, but I didn't want to go tweaking logic unrelated to the PR. I think I must have decided that committing a nil proposal at least warranted a warning though.